cmd/dockerd/daemon.go

```diff
@@ -462,7 +462,7 @@ func initRouter(opts routerOptions) {
 	routers := []router.Router{
 		// we need to add the checkpoint router before the container router or the DELETE gets masked
 		checkpointrouter.NewRouter(opts.daemon, decoder),
-		container.NewRouter(opts.daemon, decoder),
+		container.NewRouter(opts.daemon, decoder, opts.daemon.RawSysInfo(true).CgroupUnified),
```

Comment:
We already pass `opts.daemon` here, should we just use that inside `NewRouter` ?

Reply:
I'm not sure. It might not be consistent with router `Backend` interfaces.

Reply:
Please see my suggestion above. If you don't like it, at least use `cgroups.IfCgroupUnifiedMode()` here, as RawSysInfo() is relatively expensive call and we only use a tiny bit of info it provides here.

Reply:
updated